pkg/cmd/server/origin/master.go

```diff
@@ -159,7 +159,7 @@ func (c *MasterConfig) newAssetServerHandler(genericConfig *apiserver.Config) (h
 		return http.NotFoundHandler(), nil
 	}
 
-	config, err := assetapiserver.NewAssetServerConfig(*c.Options.AssetConfig)
+	config, err := assetapiserver.NewAssetServerConfig(*c.Options.AssetConfig, genericConfig.SecureServingInfo.Listener)
```

Comment:
are we passing the same listener into a bunch of server configs? That doesn't seem valid at all... only one thing can accept a connecting from the listener and handle the http request.

Reply:
+1

We can extract Address+Network from it and use that elsewhere.